### PR TITLE
[replxx] Add new port

### PR DESCRIPTION
**Pull request by kreuzerkrieg**
- What does your PR fix?
replxx new port
- Which triplets are supported/not supported?
All except UWP
- Have you updated the CI baseline?
Yes
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/replxx/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: replxx
+Version: 2019-09-22
+Homepage: https://github.com/AmokHuginnsson/replxx
+Description: A small, portable GNU readline replacement for Linux, Windows and MacOS which is capable of handling UTF-8 characters.
```
Comment (NancyLi1013) [inline ports/replxx/CONTROL]: Since this port doesn't support uwp, we can add this here.
`Supports: !uwp`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by kreuzerkrieg** [comment]
> Commenter does not have sufficient privileges for PR 10052 in repo microsoft/vcpkg

:(
bad azure

**Review by NancyLi1013 — APPROVED** [review]

**Comment by JackBoosY** [comment]
@kreuzerkrieg Just because the member has permission to rerun the test : )

File: ports/replxx/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
+The package replxx is compatible with built-in CMake targets:
+
+    FIND_PACKAGE(replxx REQUIRED)
+    TARGET_LINK_LIBRARIES(main PRIVATE REPLXX::replxx) for linkage
```
Comment (JackBoosY) [inline ports/replxx/usage]: Could you remove `for linkage`?
Usually, the user copy these 2 lines to CMakeLists.txt.

File: ports/replxx/replxxConfig.cmake [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+                          INTERFACE_INCLUDE_DIRECTORIES "${INCLUDE_DIR}"
+                          )
+    set(${NAMESPACE}_${LIB_NAME}_FOUND 1)
+endfunction()
+
+get_filename_component(ROOT "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(ROOT "${ROOT}" PATH)
+get_filename_component(ROOT "${ROOT}" PATH)
+
+set_library_target("REPLXX" "replxx" "${ROOT}/debug/lib/libreplxx-d.a" "${ROOT}/lib/libreplxx.a" "${ROOT}/include/")
```
Comment (JackBoosY) [inline ports/replxx/replxxConfig.cmake]: If this port support Windows or OSX, I think `.a` is not suitable here.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/replxx/usage (hunk 2 above)
Comment (kreuzerkrieg) [reply ports/replxx/usage]: Done

File: ports/replxx/replxxConfig.cmake (hunk 3 above)
Comment (kreuzerkrieg) [reply ports/replxx/replxxConfig.cmake]: Changed for Win
As of OSX, shouldnt it be `.a`? afterall it is BSD based OS

**Comment by NancyLi1013** [comment]
Hi @kreuzerkrieg 

The errors like this:
```
Performing C++ SOURCE FILE Test COMPILER_HAS_DEPRECATED_ATTR failed with the following output:
Change Dir: E:/10052/vcpkg/buildtrees/replxx/src/7827007a8c-9c9af6846f/build/CMakeFiles/CMakeTmp

Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_df6f8 
[1/2] Building CXX object CMakeFiles\cmTC_df6f8.dir\src.cxx.obj

FAILED: CMakeFiles/cmTC_df6f8.dir/src.cxx.obj 

C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x86\cl.exe  /nologo /TP   /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  -DCOMPILER_HAS_DEPRECATED_ATTR /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\cmTC_df6f8.dir\src.cxx.obj /FdCMakeFiles\cmTC_df6f8.dir\ /FS -c src.cxx

src.cxx(1): error C2065: '__deprecated__': undeclared identifier
src.cxx(1): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
src.cxx(1): error C2448: '__attribute__': function-style initializer appears to be a function definition
ninja: build stopped: subcommand failed.

Source file was:
__attribute__((__deprecated__)) int somefunc() { return 0; }
    int main() { return somefunc();}
Performing C++ SOURCE FILE Test COMPILER_HAS_DEPRECATED_ATTR failed with the following output:
Change Dir: E:/10052/vcpkg/buildtrees/replxx/src/7827007a8c-9c9af6846f/build/CMakeFiles/CMakeTmp
```
This info is from my local.
 This port can build successfully on `x86-windows` triplet, but we still can get this error in
`E:/10052/vcpkg/buildtrees/replxx/src/7827007a8c-9c9af6846f/build/CMakeFiles/CMakeError.log` file.

Personally, it might be the root cause of CI regression.
Could you please help chect this on your local?

**Comment by kreuzerkrieg** [comment]
Checked and it worked. I have another Win machine, will check it later, AFAIR it has older VS, like 2017, maybe it will repro there

**Comment by kreuzerkrieg** [comment]
> c:\vcpkg\vcpkg>vcpkg.exe install replxx
> Computing installation plan...
> The following packages will be built and installed:
>     replxx[core]:x86-windows
> Starting package 1/1: replxx:x86-windows
> Building package replxx[core]:x86-windows...
> -- Downloading https://github.com/AmokHuginnsson/replxx/archive/45696c250ce39ab21dedeea962b94d7827007a8c.tar.gz...
> -- Extracting source C:/vcpkg/vcpkg/downloads/AmokHuginnsson-replxx-45696c250ce39ab21dedeea962b94d7827007a8c.tar.gz
> -- Using source at C:/vcpkg/vcpkg/buildtrees/replxx/src/7827007a8c-9c9af6846f
> -- Configuring x86-windows
> -- Building x86-windows-dbg
> -- Building x86-windows-rel
> -- Installing: C:/vcpkg/vcpkg/packages/replxx_x86-windows/share/replxx/copyright
> -- Installing: C:/vcpkg/vcpkg/packages/replxx_x86-windows/share/replxx/replxxConfig.cmake
> -- Performing post-build validation
> -- Performing post-build validation done
> Building package replxx[core]:x86-windows... done
> Installing package replxx[core]:x86-windows...
> Installing package replxx[core]:x86-windows... done
> Elapsed time for package replxx:x86-windows: 43.71 s
> 
> Total elapsed time: 43.71 s
> 
> 
> c:\vcpkg\vcpkg>vcpkg.exe install replxx:x64-windows
> Computing installation plan...
> The following packages will be built and installed:
>     replxx[core]:x64-windows
> Starting package 1/1: replxx:x64-windows
> Building package replxx[core]:x64-windows...
> -- Using cached C:/vcpkg/vcpkg/downloads/AmokHuginnsson-replxx-45696c250ce39ab21dedeea962b94d7827007a8c.tar.gz
> -- Using source at C:/vcpkg/vcpkg/buildtrees/replxx/src/7827007a8c-9c9af6846f
> -- Configuring x64-windows
> -- Building x64-windows-dbg
> -- Building x64-windows-rel
> -- Installing: C:/vcpkg/vcpkg/packages/replxx_x64-windows/share/replxx/copyright
> -- Installing: C:/vcpkg/vcpkg/packages/replxx_x64-windows/share/replxx/replxxConfig.cmake
> -- Performing post-build validation
> -- Performing post-build validation done
> Building package replxx[core]:x64-windows... done
> Installing package replxx[core]:x64-windows...
> Installing package replxx[core]:x64-windows... done
> Elapsed time for package replxx:x64-windows: 34.84 s
> 
> Total elapsed time: 34.84 s


Not sure what else I can do on my side. I dont even fully understand what fails in your CI, looks like it is a CMake internal compilation of their internal cpp to check for compiler features. Looks like the MSVC miserably fails to detect a CXX feature, sounds like mismatch between MSVC and CMAKE versions. [This ](https://stackoverflow.com/questions/47135488/performing-test-compiler-has-deprecated-attr-failed)may be related.
Sounds like Gor Nishanov needed here :)

**Comment by JackBoosY** [comment]
@strega-nil Could you please have a look?

**Comment by kreuzerkrieg** [comment]
bump

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
cc @dan-shaw

**Comment by JackBoosY** [comment]
Please resolve windows regression, get failure log [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=31536&view=artifacts&type=publishedArtifacts).

**Comment by dan-shaw** [comment]
@kreuzerkrieg We are still getting the same error as before and I'm not sure what is causing the problem. Could we set this library as failing on the baseline? If someone else has the same issue, we can fix it in another PR.

**Comment by kreuzerkrieg** [comment]
done

**Review by JackBoosY — APPROVED** [review]

**Comment by dan-shaw** [comment]
@kreuzerkrieg thanks for the PR!

**Pull request merged**